pallets/handles/src/utils/validator.rs

```diff
@@ -0,0 +1,101 @@
+//! # Handle Validator
+//!
+//! `handle_validator` provides a `HandleValidator` struct to determine the validity of a given
+//! user handle.
+use sp_std::prelude::Vec;
+
+/// A validator for user handles.
+///
+/// Given a String user handle as input, determines if the handle is valid.
+pub struct HandleValidator<'string_lifetime> {
+	reserved_handles: Vec<&'string_lifetime str>,
+	blocked_characters: Vec<char>,
+	allowed_unicode_character_ranges: Vec<(u32, u32)>,
+}
+
+impl<'string_lifetime> HandleValidator<'string_lifetime> {
+	/// Create a new, intialized `HandleValidator` instance.
+	pub fn new() -> Self {
+		// This intialization data *could* be passed into the constructor and set as a pallet constant
+		let reserved_handles: Vec<&str> = Vec::from(["admin", "everyone", "all"]);
+		let blocked_characters: Vec<char> = Vec::from(['@', '#', ':', '.', '`']);
+		let allowed_unicode_character_ranges: Vec<(u32, u32)> = Vec::from([
+			(0x0020, 0x007F), // Basic Latin
```

Comment:
do can we verify all these ranges or mappings? 🤔

Reply:
There is some lookup available to get ranges from @johnmela  must have more details

Reply:
Adding a link to the source for these values: https://www.unicode.org/charts/